libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -77,7 +77,7 @@ public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions)
 
             // Invoke CollectionPolicy delegate to further customize settings
             cosmosDbStorageOptions.ConnectionPolicyConfigurator?.Invoke(connectionPolicy);
-            _client = new DocumentClient(cosmosDbStorageOptions.CosmosDBEndpoint, cosmosDbStorageOptions.AuthKey, connectionPolicy);
+            _client = cosmosDbStorageOptions.DocumentClient ?? new DocumentClient(cosmosDbStorageOptions.CosmosDBEndpoint, cosmosDbStorageOptions.AuthKey, connectionPolicy);
```

Comment:
An issue here is that we're not setting the cosmosDBStorageOptions.Document client with the newly created client.